core/src/com/unciv/logic/automation/city/ConstructionAutomation.kt

```diff
@@ -166,7 +166,7 @@ class ConstructionAutomation(val cityConstructions: CityConstructions) {
                 && city.getCenterTile().getTilesInDistance(5).none { it.militaryUnit?.civ == civInfo })
             modifier = 5f // there's a settler just sitting here, doing nothing - BAD
 
-        if (civInfo.playerType == PlayerType.Human) modifier /= 2 // Players prefer to make their own unit choices usually
+        if (civInfo.playerType == PlayerType.Human && !GUI.getSettings().autoPlay.isAutoPlaying()) modifier /= 2 // Players prefer to make their own unit choices usually
```

Comment:
I gotta say, We have to put this check (PlayerType.Human || isAutoPlaying()) alot. Can we get a val/helper function in Civilization to shorten this?

Reply:
We need to decide if autoplay is a *game* feature or a *UI* feature.
If the former, it should be under the worldscreen
If the latter, it should be under the gameInfo somewhere
It's really not a settings thing

Reply:
> This one is for !isAutoPlaying()

I mean it's just the inverse of the typical check. ``PlayerType.Human && !isAutoplaying`` is the same as ``!PlayerType.AI && !isAutoplaying`` which is the same as just inversing the helper function that I'm requesting. And I'm mostly requesting it because it seems to me (especially after adding in the personality stuff or glancing over almost any AI code) that it comes up pretty often that we're checking whether or not a player is AI controlled, rather than if a player is actually an AI